MaxText/layers/attentions.py

```diff
@@ -159,22 +143,28 @@ def query_init(*args):
   def kv_projection(self, inputs_kv: Array, proj_name: str) -> Array:
     """Projection for Key and Value.
 
-    Args;
-      inputs_kv: inputs_kv: key/values of shape `[batch, kv_length, kv_features]`.
+    Args:
+      inputs_kv: inputs_kv: key/values of shape `[batch, kv_length,
+        num_kv_heads, kv_dim]`.
       proj_name: name of projection, `key` or `value`.
 
     Returns:
-      Projection of key or value, in shape of `[batch, kv_length, num_heads,
-      head_dim]`.
+      Projection of key or value, in shape of `[batch, kv_length, head_dim]`.
     """
+    if self.num_kv_heads == -1:
```

Comment:
Nit: Could we just make this check for anything `< 0`? It's not immediately obvious to me that this is set to -1 by default somewhere.

Reply:
ack